### PR TITLE
Raise exception in failed checkout

### LINKED ISSUE #5033: git command fails but build doesn't fail
## Details

I noticed this while reporting #5032 

* Read the Docs project URL: https://readthedocs.org/projects/test-builds/builds/
* Build URL (if applicable): https://readthedocs.org/projects/test-builds/builds/8319150/
* Read the Docs username (if applicable): humitos

## Expected Result

Fail the build completely.

## Actual Result

`git` command fails, but the build continues and passes. This means that the version built is invalid and we end up serving inconsistent docs.

captura de pantalla_2018-12-24_10-48-03

### PR DESCRIPTION
**Pull request by stsewd**
Closes #5033

### CODE REVIEW

File: readthedocs/rtd_tests/tests/test_backend.py [hunk 1]
```diff
@@ -254,6 +265,17 @@ def test_update_and_checkout(self):
         self.assertEqual(code, 0)
         self.assertTrue(exists(repo.working_dir))
 
+    def test_git_checkout_invalid_revision(self):
```
Comment (humitos) [inline readthedocs/rtd_tests/tests/test_backend.py]: The test is duplicated.

I think you moved it and forget to remove the old one.

File: readthedocs/projects/exceptions.py [hunk 2]
```diff
@@ -47,6 +47,10 @@ class RepositoryError(BuildEnvironmentError):
         'You can not have two versions with the name latest or stable.'
     )
 
+    FAILED_TO_CHECKOUT = _(
+        'Failed to checkout revision: {}'
```
Comment (humitos) [inline readthedocs/projects/exceptions.py]: Will this exception log to Sentry? It should not.

File: readthedocs/rtd_tests/tests/test_backend.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/rtd_tests/tests/test_backend.py]: It's not duplicated, this one is for mercurial, forgot to change the name p:

File: readthedocs/projects/exceptions.py (hunk 2 above)
Comment (stsewd) [reply readthedocs/projects/exceptions.py]: I don't think so. Actually I'm not sure, I haven't seen a `RepositoryError` in sentry

Comment (humitos) [reply readthedocs/projects/exceptions.py]: There are some,

https://sentry.io/read-the-docs/readthedocs-org/issues/713027767/?query=RepositoryError

There are some places where it's handled

https://github.com/rtfd/readthedocs.org/blob/d2ea32ae7c25aea609970e598bafa404c0316d38/readthedocs/projects/tasks.py#L252-L273

and there other where it's not: 

https://github.com/rtfd/readthedocs.org/blob/d2ea32ae7c25aea609970e598bafa404c0316d38/readthedocs/projects/tasks.py#L211-L215

I suppose that this is the final solution, https://github.com/rtfd/readthedocs.org/issues/3856

Comment (ericholscher) [reply readthedocs/projects/exceptions.py]: Going to merge this, and if it logs, we can handle it later.

**Pull request merged**